### PR TITLE
[CP-2035] Events related to downloading MuditaOS are stored in Matomo

### PR DESCRIPTION
**Pull request by lkowalczyk87**
Jira: [CP-2035]

**Description**

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-2035]: https://appnroll.atlassian.net/browse/CP-2035?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by OskarMichalkiewicz — APPROVED** [review]

File: packages/app/src/analytic-data-tracker/helpers/track-os-download.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  return
+}
+
+export const trackOsDownload = async (
+  options: TrackOsDownloadOptions
+): Promise<void> => {
+  const { version, environment, product } = options
+
+  const e_a = version
```
Comment (dkarski) [inline packages/app/src/analytic-data-tracker/helpers/track-os-download.ts:38]: 🟡 
It looks like the logic of parse in the latest version has been omitted. This logic is important to collect the root of downloads during the sequence update.

You should send the dedicated prefix with a raw version when the version is the latest. Instead 
`latest` should be `latest - ${releaseVersion}`.

Maybe you included it and I can't read it from the code? 🤔

File: packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.test.ts [hunk 2]
```diff
@@ -128,13 +132,44 @@ describe("`AnalyticDataTrackerService`", () => {
...
+        axiosInstance
+      )
+      subject.setExternalUsageDevice(true)
+      axiosMock.onPost(apiUrl).reply(200)
+      const response = await subject.track({})
+      expect(response).not.toEqual(undefined)
+
+      subject.setExternalUsageDevice(false)
+      const response2 = await subject.track({})
+      expect(response2).toEqual(undefined)
```
Comment (dkarski) [inline packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.test.ts:161]: How does this unit test check that the flag is set to false when the body of the logic seems to look similar?

File: packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.ts [hunk 3]
```diff
@@ -37,9 +38,10 @@ export class AnalyticDataTrackerService implements AnalyticDataTrackerClass {
     this.siteId = options.siteId
     this.apiUrl = options.apiUrl
     this.trackingEnabled = options.trackingEnabled ?? true
+    this.externalUsageDevice = false
   }
   public async track(event: TrackEvent): Promise<AxiosResponse | undefined> {
-    if (!this.trackingEnabled) {
+    if (!this.trackingEnabled || !this.externalUsageDevice) {
```
Comment (dkarski) [inline packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.ts:44]: 🟡 
This check excludes the collection of all events, even those not related to the device. Let's see how to improve this to include compatibility for non-device events (like `CenterVersion`) 😉

File: packages/app/src/device/listeners/register-external-usage-device.listner.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  ipcMain.answerRenderer(
+    ExternalDataApiEvent.GetExternalUsageDevice,
+    (serialNumber: string) => {
+      return client.getExternalUsageDevice(serialNumber)
+    }
+  )
+}
+
+export default registerExternalUsageDevice
```
Comment (dkarski) [inline packages/app/src/device/listeners/register-external-usage-device.listner.ts:21]: 🟢 (a good candidate for small PR)
This is a deprecated approach. Please try to rewrite the `listener` to controller & service as a standalone module. I think that this request could merge with the first comment as a good candidate for a small refactor as a standalone PR.

**Review by dkarski — CHANGES_REQUESTED** [review]
It looks like the logic of parse in the latest version has been omitted. I can't approve without making sure it's included.

File: packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.test.ts (hunk 2 above)
Comment (dkarski) [reply packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.test.ts:161]: I don't see `not`. Sorry @lkowalczyk87

**Review by dkarski — APPROVED** [review]

**Review by OskarMichalkiewicz — APPROVED** [review]

**Pull request merged**